### PR TITLE
Add LatestBuildReason to ImageStatus

### PR DESCRIPTION
**Pull request by piyalibanerjee**
Related issue for more context: https://github.com/pivotal/kpack/issues/399  

Authored-by: Piyali Banerjee <pbanerjee@pivotal.io>

### CODE REVIEW

File: pkg/reconciler/image/reconcile_build.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ func (c *Reconciler) reconcileBuild(image *v1alpha1.Image, latestBuild *v1alpha1
 			BuildCounter:               nextBuildNumber,
 			BuildCacheName:             buildCacheName,
 			LatestBuildRef:             build.BuildRef(),
+			LatestBuildReason:          build.BuildReason(),
```
Comment (matthewmcnew) [inline pkg/reconciler/image/reconcile_build.go:38]: I think we need the LatestBuildReason populated when a build is not scheduled. (ConditionUnknown and ConditionFalse cases)

Comment (piyalibanerjee) [reply pkg/reconciler/image/reconcile_build.go:38]: Got it -- I'll add it to the ConditionFalse case (looks like ConditionUnknown is just a fallthrough). Thanks!

Comment (piyalibanerjee) [reply pkg/reconciler/image/reconcile_build.go:38]: @matthewmcnew - I added the LatestBuildReason to the ConditionFalse case in the newest commit.

**Comment by matthewmcnew** [comment]
@piyalibanerjee This is great! Thanks!

**Pull request merged**